grails-app/controllers/org/pih/warehouse/inventory/InventoryController.groovy

```diff
@@ -531,9 +531,18 @@ class InventoryController {
 
     def listInStock = {
         def location = Location.get(session.warehouse.id)
-        def inventoryItems = dashboardService.getInStock(location)
+        List<Category> categories = Category.findAllByIdInList(params.list("categories"))
 
-        if (params.format == "csv") {
+        // When accessing the page for the first time, the flag should be set to true
+        // Initially there not parameter button, only after running the report manually it is set
+        params.includeSubcategories = params.button ? params.includeSubcategories : true
+        if (params.includeSubcategories) {
+            categories = inventoryService.getExplodedCategories(categories)
+        }
+
+        def inventoryItems = dashboardService.getInStock(location, categories)
+
+        if (params.button == "download") {
```

Comment:
in this method and in methods below, a significant part seems to be the same for each of the methods (except `def inventoryItems`, but it could be also doable with some type of enum/flag I believe)  - maybe we could create a helper method or anything not to have to in the future refactor all of those three methods if ever necessary?